simulators/mujoco_simulator/revolve2/simulators/mujoco_simulator/_simulate_manual_scene.py

```diff
@@ -0,0 +1,93 @@
+"""Iterative Scene Simulator."""
+import logging
+
+import mujoco
+
+from revolve2.ci_group.simulation import (
+    STANDARD_CONTROL_FREQUENCY,
+    STANDARD_SIMULATION_TIMESTEP,
+)
+from revolve2.simulation.scene import Scene
+
+from ._control_interface_impl import ControlInterfaceImpl
+from ._custom_mujoco_viewer import CustomMujocoViewer, CustomMujocoViewerMode
+from ._scene_to_model import scene_to_model
+from ._simulation_state_impl import SimulationStateImpl
+
+
+def simulate_manual_scene(
+    scene: Scene,
+) -> None:
+    """
+    Simulate a scene for checking if a robot was built correctly.
+
+    :param scene: The scene to simulate.
+    """
+    logging.info("Simulating scene")
+
+    model, mapping = scene_to_model(
+        scene, STANDARD_SIMULATION_TIMESTEP, cast_shadows=False, fast_sim=False
+    )
+    data = mujoco.MjData(model)
+    viewer = CustomMujocoViewer(
+        model,
+        data,
+        mode=CustomMujocoViewerMode.MANUAL,
+    )
+    viewer._render_every_frame = (
+        False  # Private but functionality is not exposed and for now it breaks nothing.
+    )
+
+    # Compute forward dynamics without actually stepping forward in time.
+    mujoco.mj_forward(model, data)
+
+    control_interface = ControlInterfaceImpl(
+        data=data, abstraction_to_mujoco_mapping=mapping
+    )
+
+    """Here we set our values for cycling different positions."""
+    prev_position: int = (
+        0  # This is the initial idle position for all hinges (index of the positions).
+    )
+    positions: list[float] = [
+        0.0,
+        0.5,
+        1.0,
+        -0.5,
+        -1.0,
+    ]  # Those are the possible cycle positions we want.
+    target, current = (
+        0.0,
+        0.0,
+    )  # Here we can check whether we are currently on the correct position.
+    try:
+        while True:
+            simulation_state = SimulationStateImpl(
+                data=data, abstraction_to_mujoco_mapping=mapping
+            )
+            scene.handler.handle(
+                simulation_state, control_interface, STANDARD_CONTROL_FREQUENCY
+            )
+            # step simulation
+            if target != current:
+                """If we are not on the target we adjust our hinges accordingly."""
+                step = 0.0025 if target > current else -0.0025
```

Comment:
is this slow stepping(careful-like mode?) needed? why not directly set the target?

Reply:
if i directly set the target the simulation is unstable and doesnt work anymore. + the real robot wont be able to move instantly so i make the movement a bit smoother here. I will add this to the comments

Reply:
We do not have this in the other simulator. is that one unstable then too?